docs/api-reference/next/router.md

```diff
@@ -414,6 +414,47 @@ export default function MyApp({ Component, pageProps }) {
 }
 ```
 
+## Potential eslint errors
+
+Certain methods accessible on the `router` object return a Promise. If you have the eslint rule, [no-floating-promises](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md) enabled, you will need to either `void` the promise, or use an `async` function, `await` the Promise, then void the function call. **This is not applicable when the method is called from inside an `onClick` handler**.
```

Comment:
Please use ESLint instead of eslint 👍

Reply:
Should we just recommend disabling the ESLint rule for this line? Is there a benefit from this workaround? It feels kinda hacky when really it's from ESLint not understanding what we're trying to do, right?

Reply:
I think I agree with you tbh. I put this together in response to a discussion you flagged. If people are adding that listing rule it may be beneficial for them to know how to work around it in this case. 

But I agree it seems super hacky. Will refactor

Reply:
True that. This is a tricky one.